iroha/src/block_sync.rs

```diff
@@ -115,41 +127,49 @@ pub mod message {
     #[derive(Io, Decode, Encode, Debug, Clone)]
     pub enum Message {
         /// Gossip about latest block.
-        LatestBlock((Hash, PeerId)),
+        LatestBlock(Hash, PeerId),
         /// Request for blocks after the block with `Hash` for the peer with `PeerId`.
-        GetBlocksAfter((Hash, PeerId)),
-        /// The response to `GetBlocksAfter`. Contains the requested blocks.
-        ShareBlocks(Vec<ValidBlock>),
+        GetBlocksAfter(Hash, PeerId),
+        /// The response to `GetBlocksAfter`. Contains the requested blocks and the id of the peer who shared them.
+        ShareBlocks(Vec<ValidBlock>, PeerId),
     }
 
     impl Message {
         /// Handles the incoming message.
         pub async fn handle(&self, block_sync: &mut BlockSynchronizer) {
             match self {
-                Message::LatestBlock((hash, peer)) => {
+                Message::LatestBlock(hash, peer) => {
                     let latest_block_hash = block_sync.kura.read().await.latest_block_hash();
                     if *hash != latest_block_hash {
                         if let Err(err) =
-                            Message::GetBlocksAfter((latest_block_hash, block_sync.peer_id.clone()))
+                            Message::GetBlocksAfter(latest_block_hash, block_sync.peer_id.clone())
                                 .send_to(peer)
                                 .await
                         {
                             eprintln!("Failed to request blocks: {:?}", err)
                         }
                     }
                 }
-                Message::GetBlocksAfter((hash, peer)) => {
-                    //TODO: send batches of configurable size instead of all blocks
+                Message::GetBlocksAfter(hash, peer) => {
                     if let Some(blocks) = block_sync.kura.read().await.blocks_after(*hash) {
-                        if let Err(err) = Message::ShareBlocks(blocks.to_vec()).send_to(peer).await
+                        if let Some(blocks_batch) =
+                            blocks.chunks(block_sync.batch_size as usize).next()
```

Comment:
Because `chunks` will [panic](https://doc.rust-lang.org/std/primitive.slice.html#panics-2) if `batch_size` will be zero - we need to check that type of this field prevents such values.